vantage/dashboard_resource.go

```diff
@@ -27,127 +26,46 @@ func NewDashboardResource() resource.Resource {
-		Attributes: map[string]schema.Attribute{
-			"token": schema.StringAttribute{
-				Computed:            true,
-				MarkdownDescription: "Unique dashboard identifier",
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"title": schema.StringAttribute{
-				MarkdownDescription: "Title of the dashboard",
-				Required:            true,
-			},
-			"widget_tokens": schema.ListAttribute{
-				ElementType:         types.StringType,
-				MarkdownDescription: "Tokens for widgets present in the dashboard. Currently only cost report tokens are supported.",
-				Required:            true,
-			},
-			"date_bin": schema.StringAttribute{
-				MarkdownDescription: "Determines how to group costs in the Dashboard.",
-				Optional:            true,
-				Computed:            true,
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"date_interval": schema.StringAttribute{
-				MarkdownDescription: "Determines the date range in the Dashboard. Guaranteed to be set to 'custom' if 'start_date' and 'end_date' are set.",
-				Optional:            true,
-				Computed:            true,
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"start_date": schema.StringAttribute{
-				MarkdownDescription: "The start date for the date range for CostReports in the Dashboard. ISO 8601 Formatted. Overwrites 'date_interval' if set.",
-				Optional:            true,
-				Computed:            true,
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"end_date": schema.StringAttribute{
-				MarkdownDescription: "The end date for the date range for CostReports in the Dashboard. ISO 8601 Formatted. Overwrites 'date_interval' if set.",
-				Optional:            true,
-				Computed:            true,
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"workspace_token": schema.StringAttribute{
-				MarkdownDescription: "The token for the Workspace the Dashboard is a part of.",
-				Required:            true,
-				PlanModifiers: []planmodifier.String{
-					stringplanmodifier.UseStateForUnknown(),
-				},
-			},
-			"saved_filter_tokens": schema.ListAttribute{
-				ElementType:         types.StringType,
-				MarkdownDescription: "Tokens of the saved filters used in the Dashboard.",
-				Optional:            true,
-			},
+	s := resource_dashboard.DashboardResourceSchema(ctx)
+	attrs := s.GetAttributes()
+	s.Attributes["end_date"] = schema.StringAttribute{
+		Computed:            true,
+		Optional:            true,
+		MarkdownDescription: attrs["end_date"].GetMarkdownDescription(),
+	}
+	s.Attributes["title"] = schema.StringAttribute{
+		Computed:            true,
+		Optional:            true,
+		MarkdownDescription: attrs["title"].GetMarkdownDescription(),
+	}
+	s.Attributes["token"] = schema.StringAttribute{
```

Comment:
Oh wow, it never occurred to me to do it this way. I want to change out all the places where I did now.

Reply:
Same - An experiment. I try to try at least one thing every time I dip my toe in here :: sisyphus emoji :: for making contributions here easier

Reply:
next stop: some json annotated struct <=> tfsdk struct field conversion patterns